SEMOSS/semoss-ui
# 210 restrict extension type

created SelectSettings component to handle extensions
Updated config.tsx to add extension input with SelectSettings component 
updated UploadBlock.tsx to validate file extensions

To test the feature:

Create a drag and drop app 
add upload block
Select the upload block 
go to extension field
select multiple extensions
click save.

go back to the app
click on choose file 
file type should be auto filtered by custom files(with list of extension user specified)
user should be able to upload file with extension listed on custom file


 click on custom file and change it to "All Files(*.*)"
select the file type that is not listed on Custom file
user should not be able to upload it
